src/devices/Hmc5883l/Hmc5883l.cs

```diff
@@ -120,5 +138,30 @@ public void Dispose()
                 _sensor = null;
             }
         }
+
+        /// <summary>
+        /// Reads device statuses.
+        /// </summary>
+        /// <returns>Device statuses</returns>
+        private IList<Status> GetStatus()
+        {
+            var result = new List<Status>();
+            _sensor.WriteByte((byte)Register.HMC_STATUS_REG_ADDR);
+            byte status = _sensor.ReadByte();
+
+            if ((status & 0b_0000_0001) != 0) {
```

Comment:
this might be useful: https://docs.microsoft.com/en-us/dotnet/api/system.enum.hasflag?view=netcore-2.1

Reply:
also if status is just `[Flags]` then you don't need the list at all and can just `(Status)status`

Reply:
Fixed